### DISCUSSION TITLE
AUCROC prediction

### DISCUSSION THREAD

**Discussion by hellorp1990** [sebp/scikit-survival#269] [Show and tell]
Hi, I am new to survival analysis. I was working through the examples in the documentation and I came across an example with Veterans Lung Cancer

https://scikit-survival.readthedocs.io/en/stable/user_guide/evaluating-survival-models.html

The code ran if I use the same sample shown there. But now when I changed the validation split to 0.5 instead of 0.2, it's showing an error during the AUC calculation (cumulative_dynamic_auc) function. 

**"ValueError: time must be smaller than largest observed time point: 991.0"**

I believe I am missing something in the following line

**va_times = np.arange(8, 184, 7)**, so I tried to change, 184 to 1000 or even larger and also made them smaller but the code still showed the same error. 

Can anyone point out what am I missing?

Thank you in advance

**Comment by sebp** [comment]
The Veteran's Lung Cancer dataset is quite small (137 patients). When splitting the data into half, there's too little left to estimate the censoring distribution just from the training data. You can work around it by estimating it from the entire data by replacing `va_y_train` with `va_y`:
```python
va_times = np.arange(8, 184, 7)
cph_risk_scores = cph.predict(va_x_test)
cph_auc, cph_mean_auc = cumulative_dynamic_auc(
    va_y, va_y_test, cph_risk_scores, va_times
)
```
